src/signing/mod.rs

```diff
@@ -0,0 +1,94 @@
+use std::{
+    collections::HashMap,
+    sync::{Arc, RwLock},
+};
+
+use crate::account::Account;
+use iota::Input;
+use once_cell::sync::OnceCell;
+use serde::{Deserialize, Serialize};
+use slip10::BIP32Path;
+
+mod stronghold;
+use self::stronghold::StrongholdSigner;
+
+type BoxedSigner = Box<dyn Signer + Sync + Send>;
+type Signers = Arc<RwLock<HashMap<SignerType, BoxedSigner>>>;
+static SIGNER_INSTANCES: OnceCell<Signers> = OnceCell::new();
```

Comment:
Nit but I think `SIGNERS_INSTANCE` is more appropriate as there is one instance of signers and not instances of signer.